src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/GithubDirectClient.kt

```diff
@@ -0,0 +1,36 @@
+package org.vaccineimpact.orderlyweb.security
+
+import org.pac4j.core.client.DirectClient
+import org.pac4j.core.context.HttpConstants
+import org.pac4j.core.context.WebContext
+import org.pac4j.core.credentials.TokenCredentials
+import org.pac4j.core.credentials.extractor.HeaderExtractor
+import org.pac4j.core.profile.CommonProfile
+import org.vaccineimpact.orderlyweb.models.ErrorInfo
+import org.vaccineimpact.orderlyweb.models.permissions.PermissionSet
+
+class GithubDirectClientWrapper : CredentialClientWrapper
+{
+    override val errorInfo: ErrorInfo = ErrorInfo("github-token-invalid", "GitHub token not supplied in Authorization header, or GitHub token was invalid")
+    override val client = GithubDirectClient()
+}
+
+class GithubDirectClient : DirectClient<TokenCredentials, CommonProfile>()
+{
+    override fun clientInit(context: WebContext?)
+    {
+        defaultCredentialsExtractor(HeaderExtractor(
+                HttpConstants.AUTHORIZATION_HEADER,
+                "token ", this.name))
+
+        defaultAuthenticator(GithubAuthenticator())
+
+        setAuthorizationGenerator { _, profile -> addLoginPermission(profile) }
+    }
+
+    private fun addLoginPermission(profile: CommonProfile): CommonProfile
+    {
+        profile.montaguPermissions = PermissionSet("*/can-login")
```

Comment:
I think we probably want to deprecate the can-login permission and rather assume that anyone who is authenticated can log in. But for now I'm just adding this perm to all GitHub authenticated users. The adding of permissions to a user profile will all be revised in https://vimc.myjetbrains.com/youtrack/issue/mrc-182 anyway.